src/Raven.Client/ServerWide/Tcp/TcpConnectionHeader.cs

```diff
@@ -29,10 +30,29 @@ public enum OperationTypes
         public static readonly int NumberOfRetriesForSendingTcpHeader = 2;
         public static readonly int ClusterTcpVersion = 10;
         public static readonly int HeartbeatsTcpVersion = 20;
-        public static readonly int ReplicationTcpVersion = 31;
+        public static readonly int ReplicationTcpVersion = 33;
         public static readonly int SubscriptionTcpVersion = 40;
         public static readonly int TestConnectionTcpVersion = 50;
 
+        public static bool OperationVersionSupported(OperationTypes operationType, int version)
+        {
+            if (_operationsToSupportedProtocolVersions.TryGetValue(operationType, out var supportedProtocols) == false)
+                return false;
+            return supportedProtocols.Contains(version);
+        }
+
+        private static readonly Dictionary<OperationTypes,HashSet<int>> _operationsToSupportedProtocolVersions
+        = new Dictionary<OperationTypes, HashSet<int>>
```

Comment:
Probably better to use `List<int>` here, since order matters

Reply:
Yes, i agree.